lib/ansible/modules/monitoring/zabbix/zabbix_user_group.py

```diff
@@ -0,0 +1,369 @@
+        for rights in generated['rights']:
+            new_generated_rights.append(int(rights['permission'] + rights['id']))
+
+        new_current_rights = []
+        for rights in current['rights']:
+            new_current_rights.append(int(rights['permission'] + rights['id']))
+
+        if set(new_generated_rights) != set(new_current_rights):
+            return True
+
+        return False
+
+    def create_group(self, name, debug_mode, gui_access, users_status, rights, users):
+        if self._module.check_mode:
+            self._module.exit_json(changed=True)
+        self._zapi.usergroup.create(self.generate_group_config(name, debug_mode, gui_access, users_status, rights, users))
+
+    def update_group(self, group_id, name, debug_mode, gui_access, users_status, rights, users):
+        current_config = self.dump_group(group_id)
+        generated_config = self.generate_group_config(name, debug_mode, gui_access, users_status, rights, users)
+
+        changed = self.compare_config(generated_config, current_config)
+
+        if self._module.check_mode:
+            self._module.exit_json(changed=changed)
+
+        if changed:
+            generated_config['usrgrpid'] = group_id
+            self._zapi.usergroup.update(generated_config)
+
+        return changed
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True, aliases=['url']),
+            login_user=dict(type='str', required=True),
+            login_password=dict(type='str', required=True, no_log=True),
+            http_login_user=dict(type='str', required=False, default=None),
+            http_login_password=dict(type='str', required=False, default=None, no_log=True),
+            validate_certs=dict(type='bool', required=False, default=True),
+            name=dict(type='str', required=True),
+            debug_mode=dict(type='bool', default=False, required=False),
+            gui_access=dict(default='default', required=False, choices=['default', 'internal', 'ldap', 'disabled']),
+            users_status=dict(default='enabled', required=False, choices=['enabled', 'disabled']),
+            rights=dict(
+                type='list',
+                default=[],
+                elements='dict',
+                required=False,
+                opstions=dict(
+                    host_group=dict(type='str', required=True),
+                    permission=dict(type='str', required=True, choices=['denied', 'RO', 'RW'])
+                )
+            ),
+            users=dict(
+                type='list',
+                default=[],
+                required=False,
+                elements='str'
+            ),
+            state=dict(default="present", choices=['present', 'absent', 'dump']),
+            timeout=dict(type='int', default=10)
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_ZABBIX_API:
+        module.fail_json(msg="Missing required zabbix-api module " +
```

Comment:
Consider using the common helper for missing dependency errors.
You can refer to this code as an example.

https://github.com/ansible/ansible/blob/42a831d42c17ee7d5226796fcdcac0d6fd69b1ba/lib/ansible/modules/monitoring/zabbix/zabbix_mediatype.py#L545-L546

Reply:
Fixed in the last commit